src/tokenization.test.ts

```diff
@@ -0,0 +1,262 @@
+      const tokenIdInput = document.querySelector<HTMLInputElement>(
+        'input[name="tokenId"]',
+      )
+      expect(tokenIdInput).toBeInTheDocument()
+      expect(form).toContain(tokenIdInput)
+      expect(tokenIdInput).toBeTruthy()
+    })
+  })
+  test('should be possible to return an error if form elements do not have values assigned', async () => {
+    window.HTMLFormElement.prototype.submit = () => {}
+    onSubmit.mockImplementation((event) => {
+      event.preventDefault()
+    })
+
+    const {
+      form,
+      holderNameInput,
+      cvvInput,
+      expirationDateInput,
+      numberInput,
+    } = handleFormMock()
+
+    form.setAttribute(formElementsMock.form, '')
+    form.onsubmit = onSubmit
+    form.id = 'form'
+    form.method = 'POST'
+    form.action = '/test'
+
+    holderNameInput.setAttribute(formElementsMock.holderName, '')
+    numberInput.setAttribute(formElementsMock.number, '')
+    cvvInput.setAttribute(formElementsMock.cvv, '')
+    expirationDateInput.setAttribute(formElementsMock.expirationDate, '')
+
+    document.body.appendChild(form)
+    form.appendChild(holderNameInput)
+    form.appendChild(numberInput)
+    form.appendChild(expirationDateInput)
+    form.appendChild(cvvInput)
+
+    const malgaTokenizationObject = new MalgaTokenization(
+      malgaConfigurations(false),
+    )
+
+    await waitFor(() => {
+      expect(malgaTokenizationObject.init).rejects.toThrowError()
+    })
+
+    const formElement = document.querySelector('form')
+    fireEvent.submit(formElement!)
+  })
+  test('should be possible to return an error if apiKey and clientId are passed empty', async () => {
+    window.HTMLFormElement.prototype.submit = () => {}
+    onSubmit.mockImplementation((event) => {
+      event.preventDefault()
+    })
+
+    FormForInit(onSubmit)
+
+    const malgaConfigurationsEmpty = {
+      apiKey: '',
+      clientId: '',
+    }
+
+    const malgaTokenizationObject = new MalgaTokenization(
+      malgaConfigurationsEmpty,
+    )
+
+    await waitFor(() => {
+      expect(malgaTokenizationObject.init).rejects.toThrowError()
+    })
+
+    const form = document.querySelector('form')
+    fireEvent.submit(form!)
+  })
+})
+describe('tokenize', () => {
+  beforeEach(() => {
+    document.body.innerHTML = ''
+  })
+  test('should be possible to return a not falsy value equal to production-token-id', async () => {
+    window.HTMLFormElement.prototype.submit = () => {}
+    const malgaTokenizationObject = new MalgaTokenization(
+      malgaConfigurations(false),
+    )
+
+    FormForTokenize(handleSubmit)
+
+    const form = document.querySelector('form')
+    fireEvent.submit(form!)
+
+    let tokenId: any
+
+    async function handleSubmit(event: any) {
+      event.preventDefault()
+
+      tokenId = await malgaTokenizationObject.tokenize()
+    }
+
+    await waitFor(() => {
+      expect(tokenId.tokenId).toBeTruthy()
+      expect(tokenId.tokenId).toBe('production-token-id')
+    })
+  })
```

Comment:
Acho que aqui você não precisaria criar essa `handleSubmit` para atrelar ao evento de submit do form. Acredito que dessa forma você tem o mesmo resultado e deixa o teste mais simples:

```suggestion
  test('should be possible to return a not falsy value equal to production-token-id', async () => {
    window.HTMLFormElement.prototype.submit = () => {}
    const malgaTokenizationObject = new MalgaTokenization(
      malgaConfigurations(false),
    )

    FormForTokenize()

    const form = document.querySelector('form')
    fireEvent.submit(form!)

    const { tokenId } = await malgaTokenizationObject.tokenize()

    await waitFor(() => {
      expect(tokenId).toBe('production-token-id')
    })
  })
```